src/pycram/ontology/ontology.py

```diff
@@ -0,0 +1,501 @@
+        """
+
+        # This context manager ensures all classes created here-in share the same namepsace with `self.main_ontology`
+        with self.main_ontology:
+            # Subject
+            ontology_subject_class = self.create_ontology_concept_class(subject_class_name,
+                                                                        ontology_subject_parent_class)
+
+            # Object
+            ontology_object_class = self.create_ontology_concept_class(object_class_name, ontology_object_parent_class)
+
+            # Predicate
+            ontology_predicate_class = self.create_ontology_property_class("OntologyPredicate",
+                                                                           ontology_property_parent_class)
+            ontology_predicate_class.domain = [ontology_subject_class]
+            ontology_predicate_class.range = [ontology_object_class]
+            ontology_predicate_class.python_name = predicate_name
+
+            # Inverse Predicate
+            ontology_inverse_predicate = self.create_ontology_property_class("OntologyInversePredicate",
+                                                                             ontology_inverse_property_parent_class)
+            ontology_inverse_predicate.inverse_property = ontology_predicate_class
+            ontology_inverse_predicate.python_name = inverse_predicate_name
+
+    def create_ontology_linked_designator(self, designator_name: str, designator_class: Type[DesignatorDescription],
+                                          ontology_concept_name: str,
+                                          ontology_parent_class: Optional[Type[owlready2.Thing]] = None) \
+            -> DesignatorDescription | None:
+        """
+        Create an object designator linked to a given ontology concept
+
+        :param designator_name: Designator name
+        :param designator_class: Designator class
+        :param ontology_concept_name: Ontology concept name
+        :param ontology_parent_class: Parent ontology class from which the class of designator inherits
+        :return: An object designator associated with an ontology concept
+        """
+        ontology_concept_class = self.create_ontology_concept_class(ontology_concept_name, ontology_parent_class)
+        return self.create_ontology_linked_designator_by_concept(designator_name, designator_class,
+                                                                 ontology_concept_class)
+
+    def create_ontology_linked_designator_by_concept(self, designator_name: str,
+                                                     designator_class: Type[DesignatorDescription],
+                                                     ontology_concept_class: Type[owlready2.Thing]) \
+            -> DesignatorDescription | None:
+        """
+        Create a designator that belongs to a given ontology concept class
+
+        :param designator_name: Designator name
```

Comment:
Designator name might be confusing since it implies that it is the name of the designator itself and not what the designator describe. object_name might be better suited.